src/common/snippets/src/pass/mha_tokenization.cpp

```diff
@@ -302,16 +307,15 @@ ov::snippets::pass::TokenizeMHASnippets::TokenizeMHASnippets(const SnippetsToken
             return false;
 
         if (((reshape0 == nullptr) != (reshape1 == nullptr)) ||
-             (reshape0 && reshape1 && (reshape0->get_input_shape(0) != reshape1->get_output_shape(0))))
+             (reshape0 && reshape1 && (reshape0->get_input_partial_shape(0) != reshape1->get_output_partial_shape(0))))
             return false;
 
         // Add supported operations which are between Softmax and MatMul1 to ordered_ops
         if (!update_intermediate_supported_ops(interm_op, ordered_ops, hidden_virtual_ports_count, potential_body_params_count))
             return false;
 
         const auto matmul1 = ov::as_type_ptr<ov::opset1::MatMul>(interm_op);
-        if (!matmul1 || matmul1->get_output_target_inputs(0).size() != 1 ||
```

Comment:
Why do we remove consumers count check? Because of ShapeOf subgraphs or smth else?

Reply:
Because in general MatMul1 may have several consumers - however in this case `MatMul1` should the final node of the tokenized pattern.
Before this change we couldn't tokenize these patterns, although - why not?
```
    MatMul0
    Softmax
    MatMul1
    /    |     \
Op0  Op1  Op2
```